moveit_ros/moveit_servo/test/unit_test_servo_calcs.cpp

```diff
@@ -526,8 +526,19 @@ int main(int argc, char** argv)
   // Initialize CurrentStateMonitor
   TEST_PSM->startStateMonitor();
 
-  // Get moveit parameters
-  TEST_PARAMS = getTestParameters();
+  // read parameters and store them in shared pointer to constant
+  {
+    auto servo_parameters = std::make_shared<moveit_servo::ServoParameters>();
+    if (!moveit_servo::readParameters(servo_parameters, TEST_NODE, LOGGER))
+    {
+      RCLCPP_FATAL(LOGGER, "Failed to load the servo parameters");
+      rclcpp::shutdown();
+      return 1;
+    }
+    TEST_PARAMS = servo_parameters;
+  }
```

Comment:
Here I am also doing the parameter initialization, just this time it is in a main, and when it fails I can log and terminate.  In the case above where it is in a constructor, the workaround is I have to store a variable for the return on readParmeters and later fail if that variable is false.  Generally, I think we should keep reading parameters out of constructors except in the case of test fixtures.

Reply:
very useful example